bin/lib/zopen-build

```diff
@@ -305,27 +310,38 @@ checkEnv()
     if [ "${ZOPEN_TARBALL_DEPS}x" = "x" ]; then
       printError "ZOPEN_DEPS or ZOPEN_TARBALL_DEPS needs to be defined to the ported tools this depends on"
     fi
-    deps="${ZOPEN_TARBALL_DEPS}"
+    implicitDeps="curl tar"
+    ext=${ZOPEN_TARBALL_URL##*.}
+    if [ "${ext}x" = "xzx" ]; then
+      implicitDeps="${implicitDeps} xz"
+    elif [ "${ext}x" = "gzx" ]; then
+      implicitDeps="${implicitDeps} gzip"
+    elif [ "${ext}x" = "bz2x" ]; then
+      implicitDeps="${implicitDeps} bzip2"
+    fi
+    printVerbose "Implicitly adding tarball dependencies: ${implicitDeps}"
+    ZOPEN_TARBALL_DEPS="${implicitDeps} ${ZOPEN_TARBALL_DEPS}"
   elif [ "${ZOPEN_TYPE}x" = "GITx" ]; then
     if [ "${ZOPEN_GIT_URL}x" = "x" ]; then
       printError "ZOPEN_URL or ZOPEN_GIT_URL needs to be defined to the root directory of the tool being ported"
     fi
     if [ "${ZOPEN_GIT_DEPS}x" = "x" ]; then
       printError "ZOPEN_DEPS or ZOPEN_GIT_DEPS needs to be defined to the ported tools this depends on"
     fi
-    deps="${ZOPEN_GIT_DEPS}"
+    implicitDeps="git bash"
```

Comment:
if it's a tarball, we need curl as well.
why is bash an implicit dependency? do we have infrastructure that requires bash?

Reply:
once we get 'find' and 'xargs' working in findutils, we should include them.
if we pre-req z/os open source tools 'git' (as opposed to just rocket git), we can clean up a bunch of chtag's
if we pre-req z/os open source tools 'tar', we can clean up a couple more chtag's.
Perhaps these 3 items could be new issues post this work?

Reply:
> if it's a tarball, we need curl as well. why is bash an implicit dependency? do we have infrastructure that requires bash?

curl and tar are specified as implicit deps here: https://github.com/ZOSOpenTools/utils/pull/122/files#diff-18d8a0f3b6e2204467238e541740a1e6b0092f69dcd8f83bbf3d94627f172e69R313

I added bash because some of the git commands are bash scripts. For example git submodule

Reply:
re git & chtag,  The git we use from Rocket does all the tagging.  This might just be a version thing.  I was thinking we could allow the user to give an optional version on the dep, eg "git:2.14".

Reply:
> Are any of those used by zopen? If not then I think we can skip the bash dependency.

I think at this point there are no repos with submodules, so I'll skip it